### ISSUE TITLE
Use absolute URLs in RSS template

### ISSUE THREAD

**Issue by jemus42** [tidyverse/tidyverse.org#125]
Currently, the RSS feed for the articles [looks like this](https://www.tidyverse.org/articles/index.xml), note that the links are of the format `<link>/articles/2018/03/pillar-1-2-1/</link>`. These links seem to work fine in my regular RSS reader (Reeder.app, macOS), but if you import the feeds in e.g. Slack, the links look like this: ![](https://dump.jemu.name/2018-03-xbmg80jpzpuzxzl.png) 
Unclickable and sad.

I've looked through the page's source and saw that you seem to use Hugo's default template, so I looked at that [here](https://gohugo.io/templates/rss/#the-embedded-rss-xml), which says the RSS feed should use the `.Permalink` element. So I tried building the site locally but using `baseurl = "https://tidyverse.org/` in `config.toml`, but it didn't change the RSS links. I also tried changing the `relativeurls` option, also without luck.

The RSS feed on [my blogdown blog](https://github.com/rbind/blog.jemu.name/blob/master/config.toml) does have absolute/full URLs, and I tried using the same config with `relativeurls` and `baseurl`, but I can't seem to reproduce the desired behaviour.

Long story short: I would like to see full URLs in the articles RSS feed, but I can't figure out how to PR it myself :(

**Comment by batpigandme** [comment]
HI @jemus42,

Does it have to be RSS? I use feedly, and (as with your readers) it works fine:
![image](https://user-images.githubusercontent.com/831732/36915702-a955b9c0-1e1f-11e8-945e-0e784e21d931.png)
I'll try playing around with the settings, but as a stop-gap measure, you could use something like IFTTT to send the feed to Slack.
Changing the `relativeurls` option site-wide would wreak havoc elsewhere with the other assets, but, like I said, I'm happy to see if I can get it running, I just don't know how quickly I'll be able to do so.

**Comment by batpigandme** [comment]
Here's an alternate deploy I set up for you with full RSS links. 
[https://stupefied-noether-0f17fe.netlify.com/articles/index.xml](https://stupefied-noether-0f17fe.netlify.com/articles/index.xml)
![image](https://user-images.githubusercontent.com/831732/36917197-2a89e60c-1e24-11e8-96b1-244b87ca26d8.png)

I want to do more testing before I push it to master, but this is the same content, just at a different URL.

**Comment by jemus42** [comment]
Thanks, @batpigandme, that new feed seems to work fine!

For my own curiosity's sake: What did you change to make it work? As I said I couldn't get it to work locally.

It's really just a minor thing though because as of now the only situation where the feed URLs are an issue is with the regular slack import, where I rely on the built-in RSS app and can't use a secondary IFTTT integration due to the free-tier's limit.

**Comment by batpigandme** [comment]
Base URL, but I'm not surprised you couldn't get it to work locally, as there are a lot of different components to the site config aside from blogdown.

**Comment by lcolladotor** [comment]
Hi,

I just ran into a similar issue in 2 blogdown blogs, one with `relativeurls = true` and one with `false`. In both cases, the RSS feed seems to be populated without the `baseurl`. I _fixed_ (worked around) this issue by editing the RSS feed template. In my case, I'm using `hugo-academic` so I took the original rss template at https://github.com/gcushen/hugo-academic/blob/master/layouts/rss.xml and copied it to `layouts/_default/rss.xml`. I then found all the `<link>` parts and added my base url without the last `/`. That resulted in https://github.com/lcolladotor/hugoblog/blob/master/layouts/_default/rss.xml

In my case, `layouts/_default/rss.xml` builds the RSS feed for my categories, which is what I really wanted to fix (the rstats RSS feed that I pass to RBloggers). You could also save a copy in `layouts/rss.xml` for the main RSS feed of the blogdown page, which I had to do for the blogdown blog that hast `relativeurls = true` because it's in a subdir at GitHub: http://LieberInstitute.github.io/rstatsclub/

Best,
Leo

**Comment by lcolladotor** [comment]
Actually, nvm. I think that my issue was using `serve_site()` instead of `hugo_build()`. Sorry for the noise!

**Comment by batpigandme** [comment]
@hadley, @carneybill needs this for integration with the RStudio feed reader, and (from what I can tell) it doesn't break anything (see site with `baseurl = "https://www.tidyverse.org/"` here: https://stupefied-noether-0f17fe.netlify.com/).

**Comment by hadley** [comment]
Sure. Do you want to implement?
